### PR TITLE
HibernateDomainExporter and inheritance with composite-id.

### PR DESCRIPTION
**Pull request by vveloso**
Ported from pull request #1460 and issue #1459.

### CODE REVIEW

File: querydsl-codegen/src/main/java/com/querydsl/codegen/EntitySerializer.java [hunk 1]
```diff
@@ -64,18 +64,25 @@ public EntitySerializer(TypeMappings mappings, @Named("keywords") Collection<Str
         this.keywords = keywords;
     }
 
+    private boolean superTypeHasEntityFields(EntityType model) {
+        Supertype superType = model.getSuperType();
+        return null != superType && null != superType.getEntityType() && superType.getEntityType().hasEntityFields();
```
Comment (johnktims) [inline querydsl-codegen/src/main/java/com/querydsl/codegen/EntitySerializer.java]: For consistency's sake, could you please add a newline right before the second `&&`?

Comment (vveloso) [inline querydsl-codegen/src/main/java/com/querydsl/codegen/EntitySerializer.java]: Sure. Done.

**Comment by johnktims** [comment]
The code looks good now, but the commits are a bit noisy. Would you mind squashing them? Sorry for being so pedantic. :)

**Comment by vveloso** [comment]
I did the best I could. Everything looked good locally, I then had to merge & push, now it looks that there are even more commits. Sorry, I'm no Git guru...

**Comment by vveloso** [comment]
Is this what you intended? I'm no Git expert, :)

**Comment by vveloso** [comment]
That's pretty much what I did and it looks right.

The CI is failing with a "Killed" message. I don't think that it is related to this squash as the log looks right. Can you confirm that is indeed the case?

**Comment by Shredder121** [comment]
Yes, sorry about that. Occasionally Travis thinks the Scala compilation uses too much resources, so it kills it.
I'll restart it.

**Comment by johnktims** [comment]
@vveloso Sorry about that. I hadn't refreshed my browser and github was showing the original commits as well as the squashed commit.

**Comment by vveloso** [comment]
@johnktims No problem. It happened to me as well. :)

@Shredder121 It seems that CIs have minds of their own.

Thank you both.

**Comment by johnktims** [comment]
I'll just merge this since it's the same as #1460.

@vveloso Thanks again for the contribution and for taking the time to port it to 4.x.

**Pull request merged**